glean-core/rlb/src/lib.rs

```diff
@@ -541,5 +576,94 @@ pub fn test_reset_glean(cfg: Configuration, client_info: ClientInfoMetrics, clea
     initialize(cfg, client_info);
 }
 
+/// Sets a debug view tag.
+///
+/// When the debug view tag is set, pings are sent with a `X-Debug-ID` header with the
+/// value of the tag and are sent to the ["Ping Debug Viewer"](https://mozilla.github.io/glean/book/dev/core/internal/debug-pings.html).
+///
+/// # Arguments
+///
+/// * `tag` - A valid HTTP header value. Must match the regex: "[a-zA-Z0-9-]{1,20}".
+///
+/// # Returns
+///
+/// This will return `false` in case `tag` is not a valid tag and `true` otherwise
+/// or if the tag is set before Glean is initialized.
+pub fn set_debug_view_tag(tag: &str) -> bool {
+    if was_initialize_called() {
+        with_glean_mut(|glean| glean.set_debug_view_tag(tag))
+    } else {
+        // Glean has not been initialized yet. Cache the provided tag value.
+        if PRE_INIT_DEBUG_VIEW_TAG.get().is_none() {
+            if PRE_INIT_DEBUG_VIEW_TAG
+                .set(Mutex::new(tag.to_string()))
+                .is_err()
+            {
+                log::error!(
+                    "Unable to set the debug view tag: already set. This probably happened concurrently."
+                );
+            }
+        } else {
+            let mut lock = PRE_INIT_DEBUG_VIEW_TAG.get().unwrap().lock().unwrap();
+            *lock = tag.to_string();
+        }
```

Comment:
Isn't this just `get_or_set`, then locking and setting the value?
E.g.

```suggestion
        let m = PRE_INIT_DEBUG_VIEW_TAG.get_or_init(Default::default);
        let mut lock = m.lock().unwrap();
        *lock = tag.to_string();
```